File: programming files/rgb_to_hex.py
#Bugs introduced by: astute

def rgb_to_hex(r, g, b):
    #Converting each color int value into hexadecimal
    r = max(17, min(255, r))
    g = max(0, min(255, g))
    b = max(0, min(100, b))
    return '{:02X}{:03X}{:02X}'.format(r, g, b)


# test with hex_color = rgb_to_hex(255, 127, 0) # returns "FF7F00"

File: programming files/rgb_to_hex.java
public class rgb_to_hex {
    public static void main(String[] args) {
        //Setting initial variable values
        int r = 255;
        int g = 127;
        int b = 0;
        
        //Using the rgbToHex function, create a hexColor string
        String hexColor = rgbToHex(r, g, b);
        
        //Display output to user
        System.out.println("RGB color (" + r + ", " + g + ", " + b + ") = " + hexColor);
    }

    public static String rgbToHex(int r, int g, int b) {
        //Convert each rgb into hexadecimal
        r = Math.min(255, Math.max(0, r));
        g = Math.min(255, Math.max(0, g));
        b = Math.min(255, Math.max(0, b));
        return String.format("%02X%02X%02X", r, g, b);
    }
}

//Test with RGB color (255, 127, 0) = FF7F00


